tmpfile/tmpfile.go

```diff
@@ -0,0 +1,147 @@
+package tmpfile
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/odeke-em/go-uuid"
+)
+
+type readSeekerAt interface {
+	io.ReadSeeker
+	ReadAt(b []byte, off int64) (int, error)
+}
+
+type readSeekerAtWriteCloser interface {
+	readSeekerAt
+	io.WriteCloser
+}
+
+type statInfoer interface {
+	Stat() (os.FileInfo, error)
+}
+
+type readSeekerCloseWriterStatInfoer interface {
+	readSeekerAtWriteCloser
+	statInfoer
+}
+
+var _ readSeekerCloseWriterStatInfoer = &TmpFile{}
+
+type TmpFile struct {
+	tmpf          *os.File
+	dir           string
+	path          string
+	doneOnce      sync.Once
+	ownCreatedDir bool
+}
+
+func (tf TmpFile) Done() error {
+	var err error
+	tf.doneOnce.Do(func() {
+		_ = tf.tmpf.Close()
+		targetPath := tf.tmpf.Name()
+		if tf.ownCreatedDir {
+			// Since we are removing the entire dir,
+			// any file in the dir will be cleaned out.
+			targetPath = tf.dir
```

Comment:
I apologize, but can you point to the line where dir gets set on ttmf creation, I've had a whiskey so I could be missing it.